File: code_combat/7_Sarven_Desert/343-Alchemic_Power/alchemicPower.js
// Wait for alchemist's commands to fetch potions.

// The event handler for the pet's event "hear".
function onHear(event) {
    // Find the nearest potion.
    var potion = pet.findNearestByType("potion");
    var message = event.message;
    // If the event's message is "Fetch"
    if (message == "Fetch") {
        // Have the pet fetch the potion.
        pet.fetch(potion);
    }
    // Else (for any other messages):
    else {
        // Use pet.moveXY to return the pet to the red mark.
        pet.moveXY(54, 34);
    }
}

pet.on("hear", onHear);

// You don't have to change the code below.
while(true) {
    var enemies = hero.findEnemies();

    // Fix for weak hero
    var max = 0;
    var strongest = null;
    for (var i = 0; i < enemies.length; i++) {
        var enemy = enemies[i];
        if (enemy.health > max) {
            max = enemy.health;
            strongest = enemy;
        }
    }
    if (strongest) {
        while (strongest.health > 0)
            hero.attack(strongest);
    }
    else {
        hero.moveXY(40, 34);
    }
}

File: code_combat/7_Sarven_Desert/343-Alchemic_Power/alchemic_power.py
def onHear(event):
    potion = pet.findNearestByType("potion")
    message = event.message
    if message == "Fetch":
        pet.fetch(potion)
    else:
        pet.moveXY(54, 34)

pet.on("hear", onHear);

while True:
    enemies = hero.findEnemies()

    # Fix for weak hero
    maxim = 0
    strongest = None
    for enemy in enemies:
        if enemy.health > maxim:
            maxim = enemy.health
            strongest = enemy
    if strongest:
        while strongest.health > 0:
            hero.attack(strongest)
    else:
        hero.moveXY(40, 34)
